### PR TITLE
Data Selector Sample

### LINKED ISSUE #2924: Pivot Grid sample basic with pivot data selector

### PR DESCRIPTION
**Pull request by 3phase**
Closes #2924

### CODE REVIEW

File: src/app/pivot-grid/pivot-datasource/pivot-datasource-sample.component.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    templateUrl: './pivot-datasource-sample.component.html'
+})
+export class PivotDatasourceSampleComponent {
+    public data = DATA_TO_ANALYZE;
+    public pivotConfigHierarchy: IPivotConfiguration = {
+        columns: [
+
+            {
+                memberName: 'Region',
+                memberFunction: (data) => data.Region,
```
Comment (dkamburov) [inline src/app/pivot-grid/pivot-datasource/pivot-datasource-sample.component.ts:18]: I don't think that the `memberFunction` is needed for this scenario

**Comment by MayaKirova** [comment]
The pivot items have a lower zindex than the grid so when you drag them near the grid they disappear behind it:
list-items-zindex

I can't reproduce the same in the dev samples, so there's probably some styling in the demos that causes this.

**Review by MayaKirova — DISMISSED** [review]

**Comment by dkamburov** [comment]
@3phase The height of the grid should match the data selector, also there is no live-editing configuration for the sample

**Comment by MayaKirova** [comment]
@3phase @dkamburov 

- [ ] The sample does not fit in the size available in the docs. Either it should be smaller or the size in the docs should be updated to match:

image

- [ ] Pivot selector z-index here seems to be lower than the grid. 

image

- [ ] We should disable column selection similar to the features sample, since it's not a pivot specific feature.

- [ ] The current way the dimensions are organized will always create empty values since specific Dates are grouped in the rows while the Years are grouped in the columns. So specific dates will always have a whole column section with no values (i.e. all 2019 dates will have empty value when you scroll to the 2018 group and vice versa).
image

It would make more sense to group date related values in one dimension so that you see meaningful aggregations for it.
For example rows can contain some date hierarchy (year -> month -> date), and everything else can be grouped in the columns or vice versa.

**Review by MayaKirova — CHANGES_REQUESTED** [review]
@3phase @dkamburov 

- [x] The sample does not fit in the size available in the docs. Either it should be smaller or the size in the docs should be updated to match:

image

- [ ] Pivot selector z-index here seems to be lower than the grid. 

image

- [x] We should disable column selection similar to the features sample, since it's not a pivot specific feature.

- [x] The current way the dimensions are organized will always create empty values since specific Dates are grouped in the rows while the Years are grouped in the columns. So specific dates will always have a whole column section with no values (i.e. all 2019 dates will have empty value when you scroll to the 2018 group and vice versa).
image

It would make more sense to group date related values in one dimension so that you see meaningful aggregations for it.
For example rows can contain some date hierarchy (year -> month -> date), and everything else can be grouped in the columns or vice versa.

**Comment by dkamburov** [comment]
The size of sample in topic is updated in - https://github.com/IgniteUI/igniteui-docfx/pull/3130

**Comment by 3phase** [comment]
@MayaKirova the requested changes are applied

**Comment by dkamburov** [comment]
@3phase 
The filters from data selector are cut off
image

**Review by MayaKirova — APPROVED** [review]

**Pull request merged**